File: src/rice_cooker.py
class RiceCooker:
    def __init__(self):
        self.is_plugged = False
        self.is_empty = True
        self.is_cooking = False

rice_cooker = RiceCooker()

def plug_in(status):
    if status:
        return [True, ValueError("Rice Cooker est déjà branché")]
    return [True, None]

def unplug(is_plugged, is_cooking):
    if not is_plugged:
        return [True, ValueError("Rice Cooker est déjà débranché")]
    elif is_cooking:
        return [True, ValueError("Rice Cooker est déjà en cours d'utilisation")]
    return [False, None]

def switch_on(is_cooking, is_empty, is_plugged):
    if is_cooking:
        return [True, ValueError("Rice Cooker est déjà allumé")]
    elif is_empty:
        return [False, ValueError("Rice Cooker est déjà vide")]
    elif not is_plugged:
        return [False, ValueError("Rice Cooker est déjà débranché")]
    return [True, None]

def switch_off(is_cooking):
    if not is_cooking:
        return [False, ValueError("Rice Cooker est déjà éteint")]
    return [False, None]

def put_something(is_empty):
    if not is_empty:
        return [False, ValueError("Rice cooker est déjà rempli")]
    return [False, None]

def empty(is_empty, is_cooking):
    if is_empty:
        return [True, ValueError("Rice cooker est déjà vide")]
    elif is_cooking:
        return [False, ValueError("Rice cooker st déjà en cours de cuisson")]
    return [True, None]


while True:
    print("\nRICE COOKER")
    print("------------------------")
    print("\nMenu:\n")
    print("1- Brancher")
    print("2- Mettre du riz")
    print("3- Allumer")
    print("4- Eteindre")
    print("5- Vider")
    print("6- Débrancher")
    print("7- Quitter\n")

    choice = input("Que souhaitez-vous faire ?: ")

    if choice == "1":
        res, err = plug_in(rice_cooker.is_plugged)
        rice_cooker.is_plugged = res
        if err:
            print("Error: ", err)
        else:
            print("\nRice Cooker branché\n")
    elif choice == "2":
        res, err = put_something(rice_cooker.is_empty)
        rice_cooker.is_empty = res
        if err:
            print("Error: ", err)
        else:
            print("\nRice Cooker prêt!\n")
    elif choice == "3":
        res, err = switch_on(rice_cooker.is_cooking, rice_cooker.is_empty, rice_cooker.is_plugged)
        rice_cooker.is_cooking = res
        if err:
            print("Error: ", err)
        else:
            print("\nRice Cooker allumé!\n")
    elif choice == "4":
        res, err = switch_off(rice_cooker.is_cooking)
        rice_cooker.is_cooking = res
        if err:
            print("Error: ", err)
        else:
            print("\nRice Cooker éteint!\n")
    elif choice == "5":
        res, err = empty(rice_cooker.is_empty, rice_cooker.is_cooking)
        rice_cooker.is_empty = res
        if err:
            print("Error: ", err)
        else:
            print("\n Votre rice cooker est vide, mettez quelque chose à cuire!\n")
    elif choice == "6":
        res, err = unplug(rice_cooker.is_plugged, rice_cooker.is_cooking)
        rice_cooker.is_plugged = res
        if err:
            print("Error: ", err)
        else:
            print("\nRice Cooker débranché!\n")
    elif choice == "7":
        print("Au revoir !!")
        break
    else:
        print("")

File: src/RiceCooker.js
const prompt = require('prompt-sync')();
const myModule = require('./service/module');

class RiceCooker {
  constructor() {
    this.isPluged = false;
    this.isEmpty = true;
    this.isCooking = false;
  }
}

const riceCooker = new RiceCooker();

while (true) {
  console.log("\nRICE COOKER");
  console.log("------------------------");
  console.log("\nMenu:\n");
  console.log("1- Brancher");
  console.log("2- Mettre du riz");
  console.log("3- Allumer");
  console.log("4- Eteindre");
  console.log("5- Vider");
  console.log("6- Débrancher");
  console.log("7- Quitter\n");

  const choice = prompt("Que souhaitez-vous faire ?: ");

  if (choice === "1") {
    const [res, err] = myModule.PlugIn(riceCooker.isPluged);
    riceCooker.isPluged = res;
    if (err) {
      console.log("Error: ", err);
    } else {
      console.log("\nRice Cooker branché\n");
    }
  } else if (choice === "2") {
    const [res, err] = myModule.PutSomething(riceCooker.isEmpty);
    riceCooker.isEmpty = res;
    if (err) {
      console.log("Error: ", err);
    } else {
      console.log("\nRice Cooker prêt!\n");
    }
  } else if (choice === "3") {
    const [res, err] = myModule.SwitchOn(
      riceCooker.isCooking,
      riceCooker.isEmpty,
      riceCooker.isPluged
    );
    riceCooker.isCooking = res;
    if (err) {
      console.log("Error: ", err);
    } else {
      console.log("\nRice Cooker allumé!\n");
    }
  } else if (choice === "4") {
    const [res, err] = myModule.SwitchOff(riceCooker.isCooking);
    riceCooker.isCooking = res;
    if (err) {
      console.log("Error: ", err);
    } else {
      console.log("\nRice Cooker éteint!\n");
    }
  } else if (choice === "5") {
    const [res, err] = myModule.Empty(riceCooker.isEmpty, riceCooker.isCooking);
    riceCooker.isEmpty = res;
    if (err) {
      console.log("Error: ", err);
    } else {
      console.log("\n Votre rice cooker est vide, mettez quelque chose à cuire!\n");
    }
  } else if (choice === "6") {
    const [res, err] = myModule.Unplug(riceCooker.isPluged, riceCooker.isCooking);
    riceCooker.isPluged = res;
    if (err) {
      console.log("Error: ", err);
    } else {
      console.log("\nRice Cooker débranché!\n");
    }
  } else if (choice === "7") {
    console.log("Au revoir !!");
    break;
  } else {
    console.log("");
  }
}